### PR TITLE
Fix label tag expression toString function

### PR DESCRIPTION
**Pull request by yixinglu**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [X] enhancement

#### Description:

It's confusing for the following expression string, we should not delete the leading character for all expression.

```
-.n1.player.age
```

## How do you solve it?

keep the `$` character for the `$-`/`$$`/`$^` expressions.

In addition, more tests for PushFilterDownHashInnerJoinRule introduced in #4956 

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [x] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by Aiee — APPROVED** [review]

**Comment by Sophie-Xie** [comment]
@yixinglu plz fix the error of format.

File: src/graph/service/QueryInstance.cpp [hunk 1]
```diff
@@ -93,7 +93,8 @@ Status QueryInstance::validateAndOptimize() {
   // Optimize the query, and get the execution plan. We should not pass the optimizer errors to user
   // since the message is often not easy to understand. Logging them is enough.
   if (auto status = findBestPlan(); !status.ok()) {
-    LOG(ERROR) << "Error found in optimization stage: " << status.message();
+    LOG(ERROR) << "Error found in optimization stage for query: " << rctx->query()
+               << ", error: " << status.message();
     return Status::Error(
         "There are some errors found in optimizer, "
         "please contact to the admin to learn more details");
```
Comment (czpmango) [inline src/graph/service/QueryInstance.cpp:100]: The error message on line 100 is not appropriate, it does not prevent the error message from being passed to the user, and it is more confusing than the original error message.

File: src/common/expression/PropertyExpression.cpp [hunk 2]
```diff
@@ -209,8 +209,20 @@ void LabelTagPropertyExpression::accept(ExprVisitor* visitor) {
...
+    labelStr = label_->toString();
+    // Remove the leading '$' character for variable except '$-/$$/$^'
+    if (labelStr.find(kInputRef) != 0 && labelStr.find(kSrcRef) != 0 &&
+        labelStr.find(kDstRef) != 0) {
+      labelStr.erase(0, 1);
+    }
+    labelStr += ".";
+  }
+  if (!sym_.empty()) {
+    labelStr += sym_ + ".";
```
Comment (czpmango) [inline src/common/expression/PropertyExpression.cpp:223]: Better to use `:`.

**Review by czpmango — APPROVED** [review]

File: src/graph/service/QueryInstance.cpp (hunk 1 above)
Comment (yixinglu) [reply src/graph/service/QueryInstance.cpp:100]: Actually, I am not sure what's the better way to handle the optimizer errors. I need to open an issue about this and research how to do that in other db.

File: src/common/expression/PropertyExpression.cpp (hunk 2 above)
Comment (yixinglu) [reply src/common/expression/PropertyExpression.cpp:223]: like the following format?

```
$-:player.age
$$:player.name
v:player.name
```

File: src/graph/service/QueryInstance.cpp (hunk 1 above)
Comment (yixinglu) [reply src/graph/service/QueryInstance.cpp:100]: #5033

**Review by jievince — APPROVED** [review]

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**